python/amici/petab_objective.py

```diff
@@ -159,10 +178,116 @@ def simulate_petab(
+        PEtab parameter mapping to condition-specific
+            simulation parameters
+    :param edatas:
+        Experimental data used for simulation.
+    :param petab_scale:
+        Return sensitivities on the PEtab parameter scales
+    :param petab_problem:
+        The PEtab problem that defines the parameter scales.
+
+    :return:
+        Aggregated likelihood sensitivities.
+    """
+    accumulated_sllh = {}
+    model_parameter_ids = amici_model.getParameterIds()
+
+    if petab_scale and petab_problem is None:
+        raise ValueError(
+            'Please provide a PEtab problem if scaled SLLH is desired, or set '
+            '`petab_scale==False`.'
+        )
+
+    # Check for issues in all condition simulation results.
+    for rdata in rdatas:
+        # Condition failed during simulation.
+        if rdata['status'] != amici.AMICI_SUCCESS:
+            return None
+        # Condition simulation result does not provide SLLH.
+        if rdata.get('sllh', None) is None:
+            return None
+
+    for condition_parameter_mapping, edata, rdata in \
+            zip(parameter_mapping, edatas, rdatas):
+        for sllh_parameter_index, condition_parameter_sllh in \
+                enumerate(rdata['sllh']):
+            # Get PEtab parameter ID
+            # Use ExpData if it provides a parameter list, else default to
+            # Model.
+            if edata.plist is not None and edata.plist:
+                model_parameter_index = edata.plist[sllh_parameter_index]
+            else:
+                model_parameter_index = amici_model.plist(sllh_parameter_index)
+            model_parameter_id = model_parameter_ids[model_parameter_index]
+            petab_parameter_id = (
+                condition_parameter_mapping
+                .map_sim_var[model_parameter_id]
+            )
+
+            # Skip fixed parameters (?)
+            if not isinstance(petab_parameter_id, str):
+                continue
+
+            # Initialize
+            if petab_parameter_id not in accumulated_sllh:
+                accumulated_sllh[petab_parameter_id] = 0
+
+            partial_parameter_sllh = condition_parameter_sllh
+            # Scale
+            if petab_scale:
+                # `ParameterMappingForCondition` objects provide the scale in
+                # terms of `petab.C` constants already, not AMICI equivalents.
+                model_parameter_scale = (
+                    condition_parameter_mapping
+                    .scale_map_sim_var[model_parameter_id]
+                )
+                petab_parameter_scale = (
+                    petab_problem
+                    .parameter_df
+                    .loc[petab_parameter_id, PARAMETER_SCALE]
+                )
+                # SLLH is (un)scaled here with methods that were written for
+                # parameters.
+                # First unscale w.r.t. model scale.
+                partial_parameter_sllh = unscale_parameter(
```

Comment:
I thought we had somewhere also a function rescale(val, scale_in, scale_out), but nvm